app/assets/javascripts/components/generic_object/custom-image-component.js

```diff
@@ -0,0 +1,89 @@
+ManageIQ.angular.app.component('customImageComponent', {
+  bindings: {
+    picture: '=',
+    newRecord: '<',
+    angularForm: '<',
+    pictureUrlPath: '@',
+    pictureUploaded: '=',
+    pictureRemove: '=',
+    pictureReset: '<',
+  },
+  controllerAs: 'vm',
+  controller: customImageComponentController,
+  templateUrl: '/static/generic_object/custom-image-component.html.haml',
+});
+
+customImageComponentController.$inject = ['$timeout'];
+
+function customImageComponentController($timeout) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.imageUploadStatus = "";
+    vm.changeImage = false;
+  };
+
+  vm.$onChanges = function() {
+    vm.changeImage = false;
+    restoreOriginalStatus();
+  };
+
+  vm.uploadClicked = function() {
+    var imageFile;
+
+    if (angular.element('#generic_object_definition_image_file')[0].files.length === 0) {
+      return;
+    }
+
+    imageFile = angular.element('#generic_object_definition_image_file')[0].files[0];
+
+    var reader = new FileReader();
+    vm.imageUploadStatus = "";
+
+    if (imageFile.type === 'image/png') {
+      vm.picture.extension = 'png';
+    } else if (imageFile.type === 'image/jpg') {
+      vm.picture.extension = 'jpg';
+    } else if (imageFile.type === 'image/jpeg') {
+      vm.picture.extension = 'jpeg';
```

Comment:
`{"error":{"kind":"bad_request","message":"Failed to update generic object definition - Validation failed: Extension must be a png, jpg, or svg","klass":"Api::BadRequestError"}}`

This has to be `jpg` as well.

And it seems like you may want to add a condition for SVG.

Reply:
Addressed in fb22a12